ceph/rados/core_workflows.py

```diff
@@ -1980,3 +1971,36 @@ def restart_daemon_services(self, daemon: str):
 
         log.info(f"Ceph Orch Service(s) {daemon_services} has been restarted")
         return True
+
+    def check_pool_pg_states(self, pool: str, disallowed_states: list):
+        """
+        the method fetches the states for the pg belonging to particular pool.
+        If any of the PG in the pool, have states that are in the disallowed states list, returns fail
+
+        Args::
+            pool: Name of the pool whose pg states need to be checked
+            disallowed_states: list of pg states, that are not allowed, if found, module returns fail
+
+        Returns::
+            Pass -> True
+            Fail -> False
+        """
+        # fetching the pool ID
+        cmd = "ceph df"
+        out = self.run_ceph_command(cmd=cmd)
+        pool_names = [entry["name"] for entry in out.get("pools", [])]
+        pool_id = out["pools"][pool_names.index(pool)]["id"]
+
+        cmd = f" ceph pg ls {pool_id}"
+        out = self.run_ceph_command(cmd=cmd)
+        for ele in out["pg_stats"]:
+            if any(key in disallowed_states for key in ele["state"].split("+")):
+                log.error(
+                    f"PG : {ele['pgid']} is in state : {ele['state']}. "
+                    "PG expected to be active+clean"
```

Comment:
Please replace with `"PG not expected to be in {key} state."`

Reply:
I believe it's better to print Expected vs present states, with PG ID.